src/wallet-metrics.ts

```diff
@@ -0,0 +1,388 @@
+import { BigNumber, ethers, Wallet } from 'ethers';
+import { uniqBy } from 'lodash';
+import { NonceManager } from '@ethersproject/experimental';
+import { parseEther } from 'ethers/lib/utils';
+import { OperationsRepository } from '@api3/operations';
+import { go } from '@api3/airnode-utilities';
+import { PROTOCOL_ID_PSP } from '@api3/operations/dist/utils/evm';
+import {
+  evm,
+  logging,
+  opsGenie,
+  promises,
+  TelemetryChainConfig,
+  OpsGenieConfig,
+  GlobalSponsor,
+} from '@api3/operations-utilities/dist';
+import { API3_XPUB } from './constants';
+import { ExtendedWalletWithMetadata, WalletStatus, WalletConfig } from './types';
+
+export const getGlobalProvider = async (
+  chains: Record<string, TelemetryChainConfig>,
+  opsGenieConfig: OpsGenieConfig,
+  id: string
+) => {
+  if (chains[id]) {
+    return chains[id].rpc;
+  }
+
+  await opsGenie.sendToOpsGenieLowLevel(
+    {
+      message: `No provider found for chain id ${id}`,
+      alias: `no-provider-found-${id}`,
+      description: `No provider found for this chain ID, please check the config/walletConfig.json file`,
+      priority: 'P2',
+    },
+    opsGenieConfig
+  );
+  return '';
+};
+
+/**
+ * Notes
+ *
+ * A WALLET_ENABLE_SEND_FUNDS env must be set to any value in production to enable actual wallet top-ups.
+ */
+
+/**
+ * Returns a provider instance based on global config and a chain name
+ * @param walletConfig the result of parsing walletConfig.json
+ * @param chainName the name of the chain
+ */
+export const getProvider = async (walletConfig: WalletConfig, chainName: string) => {
+  const chainId = evm.resolveChainId(chainName);
+  if (!chainId) {
+    await opsGenie.sendToOpsGenieLowLevel(
+      {
+        message: 'Invalid chain name',
+        alias: `invalid-chain-name-${chainName}`,
+        description: `Check that the chains specified in the ops repository and in config/walletConfig.json match`,
+        priority: 'P3',
+      },
+      walletConfig.opsGenieConfig
+    );
+  } else {
+    await opsGenie.closeOpsGenieAlertWithAlias(`invalid-chain-name-${chainName}`, walletConfig.opsGenieConfig);
+  }
+  const providerUrl = await getGlobalProvider(walletConfig.chains, walletConfig.opsGenieConfig, chainId!);
+
+  return new ethers.providers.StaticJsonRpcProvider(providerUrl, { chainId: parseInt(chainId!), name: chainName });
+};
+
+// TODO replace this when it becomes available for import in upstream
+// The version we need appears to be in packages/airnode-node/src/evm/wallet.ts
+export const deriveWalletPathFromSponsorAddress = (sponsorAddress: string, protocolId = '1') => {
```

Comment:
This was copied out of Airnode I think because the protocol IDs has changed, but if you're in the mood you can probably use an Airnode import instead now.